### ISSUE TITLE
[Bug] discord_arch_electron corrupted

### ISSUE THREAD

**Issue by vascorsd** [chaotic-aur/packages#1383]
### What happens?

Trying to reinstall the package `discord_arch_electron` and keep getting a problem with the signature.

```
Package (1)                        New Version  Net Change  Download Size

chaotic-aur/discord_arch_electron  0.0.17-3      10,96 MiB       2,11 MiB

Total Download Size:    2,11 MiB
Total Installed Size:  10,96 MiB

:: Proceed with installation? [Y/n] 
:: Retrieving packages...
 discord_arch_electron-0.0.17-3-x86_64                                2,1 MiB  2,46 MiB/s 00:01 [--------------------------------------------------------] 100%
(1/1) checking keys in keyring                                                                  [--------------------------------------------------------] 100%
(1/1) checking package integrity                                                                [--------------------------------------------------------] 100%
error: discord_arch_electron: signature from "Pedro Henrique Lara Campos <root@pedrohlc.com>" is invalid
:: File /var/cache/pacman/pkg/discord_arch_electron-0.0.17-3-x86_64.pkg.tar.zst is corrupted (invalid or corrupted package (PGP signature)).
Do you want to delete it? [Y/n] 
error: failed to commit transaction (invalid or corrupted package (PGP signature))
Errors occurred, no packages were upgraded.

```

### What is expected to happen?

The package should be correct and install as expected without giving an error about signature.

### If possible, please attach logs

```
❯ sudo pacman -Sw --debug  discord_arch_electron
debug: pacman v6.0.1 - libalpm v13.0.1
debug: config: attempting to read file /etc/pacman.conf
debug: config: new section 'options'
debug: config: CacheDir: /var/cache/pacman/pkg
debug: config: CacheDir: /var/cache/paru/pkg
debug: config: HoldPkg: pacman
debug: config: HoldPkg: glibc
debug: config: Architecture: auto
debug: config: arch: x86_64
debug: config: verbosepkglists
debug: config: chomp
debug: config: SigLevel: Required
debug: config: SigLevel: DatabaseOptional
debug: config: LocalFileSigLevel: Optional
debug: config: new section 'core'
debug: config file /etc/pacman.conf, line 77: including /etc/pacman.d/mirrorlist
debug: config: new section 'extra'
debug: config file /etc/pacman.conf, line 80: including /etc/pacman.d/mirrorlist
debug: config: new section 'community'
debug: config file /etc/pacman.conf, line 86: including /etc/pacman.d/mirrorlist
debug: config: new section 'multilib'
debug: config file /etc/pacman.conf, line 95: including /etc/pacman.d/mirrorlist
debug: config: new section 'chaotic-aur'
debug: config file /etc/pacman.conf, line 108: including /etc/pacman.d/chaotic-mirrorlist
debug: config: new section 'userrepository'
debug: config: SigLevel: Optional
debug: config: SigLevel: TrustAll
debug: config: new section 'paru-local-repo'
debug: config: SigLevel: PackageOptional
debug: config: SigLevel: DatabaseOptional
debug: config: finished parsing /etc/pacman.conf
debug: setup_libalpm called
debug: option 'logfile' = /var/log/pacman.log
debug: option 'gpgdir' = /etc/pacman.d/gnupg/
debug: option 'hookdir' = /etc/pacman.d/hooks/
debug: option 'cachedir' = /var/cache/pacman/pkg/
debug: option 'cachedir' = /var/cache/paru/pkg/
debug: registering sync database 'core'
debug: database path for tree core set to /var/lib/pacman/sync/core.db
debug: "/var/lib/pacman/sync/core.db.sig" is not readable: No such file or directory
debug: sig path /var/lib/pacman/sync/core.db.sig could not be opened
debug: got error 43 at _alpm_gpgme_checksig (../lib/libalpm/signing.c: 593) : missing PGP signature
debug: missing optional signature
debug: setting usage of 15 for core repository
debug: adding new server URL to database 'core': https://mirror.chaoticum.net/arch/core/os/x86_64
debug: adding new server URL to database 'core': https://archlinux.mailtunnel.eu/core/os/x86_64
debug: adding new server URL to database 'core': https://archlinux.thaller.ws/core/os/x86_64
debug: adding new server URL to database 'core': https://mirror.pkgbuild.com/core/os/x86_64
debug: adding new server URL to database 'core': https://mirror.cyberbits.eu/archlinux/core/os/x86_64
debug: registering sync database 'extra'
debug: database path for tree extra set to /var/lib/pacman/sync/extra.db
debug: "/var/lib/pacman/sync/extra.db.sig" is not readable: No such file or directory
debug: sig path /var/lib/pacman/sync/extra.db.sig could not be opened
debug: got error 43 at _alpm_gpgme_checksig (../lib/libalpm/signing.c: 593) : missing PGP signature
debug: missing optional signature
debug: setting usage of 15 for extra repository
debug: adding new server URL to database 'extra': https://mirror.chaoticum.net/arch/extra/os/x86_64
debug: adding new server URL to database 'extra': https://archlinux.mailtunnel.eu/extra/os/x86_64
debug: adding new server URL to database 'extra': https://archlinux.thaller.ws/extra/os/x86_64
debug: adding new server URL to database 'extra': https://mirror.pkgbuild.com/extra/os/x86_64
debug: adding new server URL to database 'extra': https://mirror.cyberbits.eu/archlinux/extra/os/x86_64
debug: registering sync database 'community'
debug: database path for tree community set to /var/lib/pacman/sync/community.db
debug: "/var/lib/pacman/sync/community.db.sig" is not readable: No such file or directory
debug: sig path /var/lib/pacman/sync/community.db.sig could not be opened
debug: got error 43 at _alpm_gpgme_checksig (../lib/libalpm/signing.c: 593) : missing PGP signature
debug: missing optional signature
debug: setting usage of 15 for community repository
debug: adding new server URL to database 'community': https://mirror.chaoticum.net/arch/community/os/x86_64
debug: adding new server URL to database 'community': https://archlinux.mailtunnel.eu/community/os/x86_64
debug: adding new server URL to database 'community': https://archlinux.thaller.ws/community/os/x86_64
debug: adding new server URL to database 'community': https://mirror.pkgbuild.com/community/os/x86_64
debug: adding new server URL to database 'community': https://mirror.cyberbits.eu/archlinux/community/os/x86_64
debug: registering sync database 'multilib'
debug: database path for tree multilib set to /var/lib/pacman/sync/multilib.db
debug: "/var/lib/pacman/sync/multilib.db.sig" is not readable: No such file or directory
debug: sig path /var/lib/pacman/sync/multilib.db.sig could not be opened
debug: got error 43 at _alpm_gpgme_checksig (../lib/libalpm/signing.c: 593) : missing PGP signature
debug: missing optional signature
debug: setting usage of 15 for multilib repository
debug: adding new server URL to database 'multilib': https://mirror.chaoticum.net/arch/multilib/os/x86_64
debug: adding new server URL to database 'multilib': https://archlinux.mailtunnel.eu/multilib/os/x86_64
debug: adding new server URL to database 'multilib': https://archlinux.thaller.ws/multilib/os/x86_64
debug: adding new server URL to database 'multilib': https://mirror.pkgbuild.com/multilib/os/x86_64
debug: adding new server URL to database 'multilib': https://mirror.cyberbits.eu/archlinux/multilib/os/x86_64
debug: registering sync database 'chaotic-aur'
debug: database path for tree chaotic-aur set to /var/lib/pacman/sync/chaotic-aur.db
debug: "/var/lib/pacman/sync/chaotic-aur.db.sig" is not readable: No such file or directory
debug: sig path /var/lib/pacman/sync/chaotic-aur.db.sig could not be opened
debug: got error 43 at _alpm_gpgme_checksig (../lib/libalpm/signing.c: 593) : missing PGP signature
debug: missing optional signature
debug: setting usage of 15 for chaotic-aur repository
debug: adding new server URL to database 'chaotic-aur': https://geo-mirror.chaotic.cx/chaotic-aur/x86_64
debug: adding new server URL to database 'chaotic-aur': https://cdn-mirror.chaotic.cx/chaotic-aur/x86_64
debug: adding new server URL to database 'chaotic-aur': https://us-ut-mirror.chaotic.cx/chaotic-aur/x86_64
debug: adding new server URL to database 'chaotic-aur': https://es-mirror.chaotic.cx/chaotic-aur/x86_64
debug: adding new server URL to database 'chaotic-aur': https://es-2-mirror.chaotic.cx/chaotic-aur/x86_64
debug: adding new server URL to database 'chaotic-aur': https://de-mirror.chaotic.cx/chaotic-aur/x86_64
debug: adding new server URL to database 'chaotic-aur': https://de-2-mirror.chaotic.cx/chaotic-aur/x86_64
debug: adding new server URL to database 'chaotic-aur': https://de-3-mirror.chaotic.cx/chaotic-aur/x86_64
debug: adding new server URL to database 'chaotic-aur': https://de-4-mirror.chaotic.cx/chaotic-aur/x86_64
debug: adding new server URL to database 'chaotic-aur': https://fr-mirror.chaotic.cx/chaotic-aur/x86_64
debug: adding new server URL to database 'chaotic-aur': https://br-mirror.chaotic.cx/chaotic-aur/x86_64
debug: adding new server URL to database 'chaotic-aur': https://kr-mirror.chaotic.cx/chaotic-aur/x86_64
debug: adding new server URL to database 'chaotic-aur': https://bg-mirror.chaotic.cx/chaotic-aur/x86_64
debug: adding new server URL to database 'chaotic-aur': https://ca-mirror.chaotic.cx/chaotic-aur/x86_64
debug: adding new server URL to database 'chaotic-aur': https://in-mirror.chaotic.cx/chaotic-aur/x86_64
debug: adding new server URL to database 'chaotic-aur': https://in-2-mirror.chaotic.cx/chaotic-aur/x86_64
debug: registering sync database 'userrepository'
debug: database path for tree userrepository set to /var/lib/pacman/sync/userrepository.db
debug: "/var/lib/pacman/sync/userrepository.db.sig" is not readable: No such file or directory
debug: sig path /var/lib/pacman/sync/userrepository.db.sig could not be opened
debug: got error 43 at _alpm_gpgme_checksig (../lib/libalpm/signing.c: 593) : missing PGP signature
debug: missing optional signature
debug: setting usage of 15 for userrepository repository
debug: adding new server URL to database 'userrepository': https://userrepository.eu
debug: registering sync database 'paru-local-repo'
debug: database path for tree paru-local-repo set to /var/lib/pacman/sync/paru-local-repo.db
debug: "/var/lib/pacman/sync/paru-local-repo.db.sig" is not readable: No such file or directory
debug: sig path /var/lib/pacman/sync/paru-local-repo.db.sig could not be opened
debug: got error 43 at _alpm_gpgme_checksig (../lib/libalpm/signing.c: 593) : missing PGP signature
debug: missing optional signature
debug: setting usage of 15 for paru-local-repo repository
debug: adding new server URL to database 'paru-local-repo': file:///var/cache/paru/pkg
debug: loading package cache for repository 'core'
debug: opening archive /var/lib/pacman/sync/core.db
debug: added 256 packages to package cache for db 'core'
debug: loading package cache for repository 'extra'
debug: opening archive /var/lib/pacman/sync/extra.db
debug: added 2968 packages to package cache for db 'extra'
debug: loading package cache for repository 'community'
debug: opening archive /var/lib/pacman/sync/community.db
debug: added 9117 packages to package cache for db 'community'
debug: loading package cache for repository 'multilib'
debug: opening archive /var/lib/pacman/sync/multilib.db
debug: added 278 packages to package cache for db 'multilib'
debug: loading package cache for repository 'chaotic-aur'
debug: opening archive /var/lib/pacman/sync/chaotic-aur.db
debug: added 2609 packages to package cache for db 'chaotic-aur'
debug: adding package 'discord_arch_electron'
debug: loading package cache for repository 'local'
debug: added 1479 packages to package cache for db 'local'
debug: adding package discord_arch_electron-0.0.17-3 to the transaction add list
resolving dependencies...
debug: resolving target's dependencies
debug: started resolving dependencies
debug: checkdeps: package discord_arch_electron-0.0.17-3
debug: finished resolving dependencies
debug: checking dependencies
debug: checkdeps: package discord_arch_electron-0.0.17-3
debug: returning error 33 from alpm_db_get_pkg (../lib/libalpm/db.c: 248) : could not find or read package
debug: found cached pkg: /var/cache/pacman/pkg/discord_arch_electron-0.0.17-3-x86_64.pkg.tar.zst
debug: setting download size 0 for pkg discord_arch_electron
debug: sorting by dependencies
debug: started sorting dependencies
debug: sorting dependencies finished
debug: returning error 33 from alpm_db_get_pkg (../lib/libalpm/db.c: 248) : could not find or read package

Package (1)                        New Version  Net Change

chaotic-aur/discord_arch_electron  0.0.17-3      10,96 MiB

Total Download Size:  0,00 MiB

:: Proceed with download? [Y/n] 
debug: using cachedir: /var/cache/pacman/pkg/
debug: found cached pkg: /var/cache/pacman/pkg/discord_arch_electron-0.0.17-3-x86_64.pkg.tar.zst
debug: found cached pkg: /var/cache/pacman/pkg/discord_arch_electron-0.0.17-3-x86_64.pkg.tar.zst.sig
checking keyring...
debug: found signature key: 3A40CB5E7E5CBC30
debug: GPGME version: 1.17.1
debug: GPGME engine info: file=/usr/bin/gpg, home=/etc/pacman.d/gnupg/
debug: looking up key 3A40CB5E7E5CBC30 locally
debug: key lookup success, key exists
checking package integrity...
debug: found cached pkg: /var/cache/pacman/pkg/discord_arch_electron-0.0.17-3-x86_64.pkg.tar.zst
debug: sig data: iQEzBAABCAAdFiEEv7E+pQfv2ttkqUSBOkDLXn5cvDAFAmIeUn8ACgkQOkDLXn5cvDDl6wf/aO+AEEWpxz6/AIuAxR2OTkwkt+mOcrdFNZ8Oe+gic72lYE8a1UDdIaPwE+N36aVJnWVnCL18L6mQoj56ZLXnr/nsyfMV6cJSiiPpIq3g4uNgVBN928d3ZpSukvPa/ev1iPWcgPERvPemhOM6w9O0F3v/AJIOe5WFNIPwdP5VqDVLRjFheXu9AQdwB/92z444hUsNi77BMYQ1itTKqSRWz+6AW/DOHW6DGUOC6vqOnXdMqYsdlOuFxwE+NeFM8cJwmnjJhGYdDQeHKn0hMUzc36zoN0Kuw0YNuZB/7W2OvzdYziTUnK69iwEtAyYb29RZVQkx+Ls9jGWzU9CfEGg2vw==
debug: checking signature for /var/cache/pacman/pkg/discord_arch_electron-0.0.17-3-x86_64.pkg.tar.zst
debug: 1 signatures returned
debug: fingerprint: 3A40CB5E7E5CBC30
debug: summary: red
debug: status: Bad signature
debug: timestamp: 0
debug: exp_timestamp: 0
debug: validity: unknown; reason: Success
debug: key: EF925EA60F33D0CB85C44AD13056513887B78AEB, Pedro Henrique Lara Campos <root@pedrohlc.com>, owner_trust marginal, disabled 0
debug: signature is not valid
error: discord_arch_electron: signature from "Pedro Henrique Lara Campos <root@pedrohlc.com>" is invalid
:: File /var/cache/pacman/pkg/discord_arch_electron-0.0.17-3-x86_64.pkg.tar.zst is corrupted (invalid or corrupted package (PGP signature)).
Do you want to delete it? [Y/n] 
error: failed to commit transaction (invalid or corrupted package (PGP signature))
Errors occurred, no packages were upgraded.
debug: unregistering database 'local'
debug: freeing package cache for repository 'local'
debug: unregistering database 'core'
debug: freeing package cache for repository 'core'
debug: unregistering database 'extra'
debug: freeing package cache for repository 'extra'
debug: unregistering database 'community'
debug: freeing package cache for repository 'community'
debug: unregistering database 'multilib'
debug: freeing package cache for repository 'multilib'
debug: unregistering database 'chaotic-aur'
debug: freeing package cache for repository 'chaotic-aur'
debug: unregistering database 'userrepository'
debug: unregistering database 'paru-local-repo'

```

### More information

Have been installing other stuff from chaotic today without any problems.

When I reached to reinstall this discord package it kept giving errors.
I tried all the recommended and ways to properly add the keys, cleaned arch keys database, made sure the server for gpg was pointed to ubuntu one, refresh the keys multiple times, forced syncing repos and redownload of the chaotic-keyring package, changed the chaotic server to directly use another server instead of the Include=chaotic-mirrors, tried a bunch of them, verified that the keys for root@pedrohlc.com are there with pacman-key --list-sigs, etc.

Basically I tried all I could think and do. I can only reach the conclusion the package is truly corrupted and can't be installed.

**Comment by dr460nf1r3** [comment]
The package has been rebuilt, please try again :)

**Comment by xiota** [comment]
What benefit does this package have over `extra/discord`?

Anyone notice the size difference?

```
$ pacman -Si discord_arch_electron | grep -E '^(Repo|Name|.*Size)'
Repository      : chaotic-aur
Name            : discord_arch_electron
Download Size   : 1585.55 KiB
Installed Size  : 8346.34 KiB

$ pacman -Si discord | grep -E '^(Repo|Name|.*Size)'
Repository      : extra
Name            : discord
Download Size   : 68.22 MiB
Installed Size  : 220.91 MiB
```

**Comment by dr460nf1r3** [comment]
It's using Archs Electron packages rather than bundling it, therefore the size goes down 👀

**Comment by xiota** [comment]
O... it's KiB?  K and M look the same after a while.
